raft/raft.go

```diff
@@ -172,6 +173,9 @@ func newRaft(id uint64, peers []uint64, election, heartbeat int, storage Storage
 	if !isHardStateEqual(hs, emptyState) {
 		r.loadState(hs)
 	}
+	if applied > 0 {
```

Comment:
if the passed in applied is not zero, a correct applied should always be greater than raftlog.firstIndex(). right?

so i prefer raftlog.applied.